### PR TITLE
Add starters for `bindings/web5_c` and `bound/golang`

### PR DESCRIPTION
**Pull request by KendallWeihe**
Marking this as a draft for now, opening for visibility

### CODE REVIEW

File: bindings/web5_c/src/crypto/key_managers/poc.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    c::free_bytes,
+    crypto::{
+        dsa::{call_sign, free_csigner},
+        jwk::{free_cjwk, CJwk},
+    },
+};
+use web5::crypto::jwk::Jwk;
+
+#[no_mangle]
+pub extern "C" fn poc_key_manager_from_foreign(manager: *const CKeyManager) {
```
Comment (nitro-neal) [inline bindings/web5_c/src/crypto/key_managers/poc.rs:12]: this is a test basically?

File: bound/golang/web5c/export.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package web5c
+
+/*
+#include <stdlib.h>
+
+#include "../../../bindings/web5_c/web5_c.h"
+*/
```
Comment (nitro-neal) [inline bound/golang/web5c/export.go:7]:
```suggestion
```

**Review by nitro-neal — APPROVED** [review]

**Comment by nitro-neal** [comment]
lots of errors in linter, maybe need to disable for bindings/web5_c or edit

File: bindings/web5_c/src/crypto/key_managers/poc.rs (hunk 1 above)
Comment (KendallWeihe) [reply bindings/web5_c/src/crypto/key_managers/poc.rs:12]: yup! it proves the injection of a foreign implementation of a key manager

File: bound/golang/web5c/export.go (hunk 2 above)
Comment (KendallWeihe) [reply bound/golang/web5c/export.go:7]: this is required, it's the way cgo works, it's called the "preamble"

**Pull request merged**